sdks/python/apache_beam/ml/transforms/base.py

```diff
@@ -230,3 +244,26 @@ def _validate_transform(self, transform):
       raise TypeError(
           'transform must be a subclass of BaseOperation. '
           'Got: %s instead.' % type(transform))
+
+
+class MLTransformMetricsUsage(beam.PTransform):
+  def __init__(self, ml_transform: MLTransform):
+    self._ml_transform = ml_transform
+    self._ml_transform._counter.inc()
+
+  def expand(self, pipeline):
+    def _increment_counters():
+      # increment for MLTransform.
+      self._ml_transform._counter.inc()
+      # increment if data processing transforms are passed.
+      transforms = (
+          self._ml_transform.transforms or
+          self._ml_transform._process_handler.transforms)
+      if transforms:
+        for transform in transforms:
+          transform.get_counter().inc()
+
+    _ = (
+        pipeline
+        | beam.Create([None])
```

Comment:
more of a understanding question - why do we create an empty pcollection here instead of directly appending to pipeline?

Reply:
In this context, we only need to update the metrics once and i find this to be the easy way to do it.  This is similar to how TFT does it. I needed the same behavior but in our MLTransform so I replicated it from https://source.corp.google.com/piper///depot/google3/third_party/py/tensorflow_transform/beam/impl.py;l=922?q=tft_mapper_